domain/src/test/java/org/oppia/android/domain/profile/ProfileManagementControllerTest.kt

```diff
@@ -167,23 +171,43 @@ class ProfileManagementControllerTest {
   }
 
   @Test
-  fun testAddProfile_addProfileWithNumberInName_checkResultIsFailure() {
+  fun testAddProfile_addProfilesWithDisallowedNames_checkResultIsFailure() {
     addTestProfiles()
     testCoroutineDispatchers.runCurrent()
 
-    profileManagementController.addProfile(
-      name = "James034",
-      pin = "321",
-      avatarImagePath = null,
-      allowDownloadAccess = false,
-      colorRgb = -10710042,
-      isAdmin = true
-    ).toLiveData().observeForever(mockUpdateResultObserver)
+    disallowedNames.forEach {
+      profileManagementController.addProfile(
+        name = it,
+        pin = "321",
+        avatarImagePath = null,
+        allowDownloadAccess = false,
+        colorRgb = -10710042,
+        isAdmin = true
+      ).toLiveData().observeForever(mockUpdateResultObserver)
+      testCoroutineDispatchers.runCurrent()
+
+      verifyUpdateFailed()
+      assertThat(updateResultCaptor.value.getErrorOrNull()).hasMessageThat()
+        .contains("$it does not contain only letters")
```

Comment:
This error seems no longer correct--we should update it.

@srushtirk do you have suggestions on what the new error message should be?

Reply:
Is this still unresolved @JishnuGoyal?

Also, I think we should make sure that there are at least a few tests for name validation in ProfileManagementControllerTest, too, such as:
- One valid name with Latin characters
- One valid name with Devanagari characters (which would fail without the new validator)
- One invalid name (with the correct presented)

What do you think?

Reply:
I will take over answering this one, since @srushtirk is away.

Let's go with this error message: "Please choose a profile name that doesn't include numbers or symbols."

(Also, @JishnuGoyal, in the future, if you don't hear back from someone within 2 days and it's blocking you, please escalate. Thanks.)

Reply:
@JishnuGoyal reminder to please address this comment.

Reply:
add_profile_error_name_only_letters still needs to be updated to have a clearer error.

Reply:
Resolved in #4565.